blockchain/skipchain/block.go

```diff
@@ -217,13 +244,17 @@ type Chain []SkipBlock
 
 // GetProof returns a verifiable proof of the latest block of the chain.
 func (c Chain) GetProof() Proof {
-	chain := c[:len(c)-1]
+	numForwardLink := int(math.Max(0, float64(len(c)-1)))
+	if len(c) == 0 {
+		return Proof{}
+	}
 
 	proof := Proof{
-		ForwardLinks: make([]ForwardLink, len(chain)),
+		GenesisBlock: c[0],
+		ForwardLinks: make([]ForwardLink, numForwardLink),
 	}
 
-	for i, block := range chain {
+	for i, block := range c[:numForwardLink] {
```

Comment:
```suggestion
	for i, block := range c {
```

Reply:
there is one forward link less than the number of blocks